src/core.jl

```diff
@@ -78,24 +78,37 @@ done(A::FixedArray, state::Integer) = length(A) < state
     :($(T.name.primary))
 end
 
-similar{FSA <: FixedVector, T}(::Type{FSA}, ::Type{T}, n::Tuple) = similar(FSA, T, n...)
-@generated function similar{FSA <: FixedVector, T}(::Type{FSA}, ::Type{T}, n::Int)
-    name = basetype(FSA)
-    :($name{n, T, $(FSA.parameters[3:end]...)})
-end
-@generated function similar{FSA <: FixedVector, T}(::Type{FSA}, ::Type{T})
-    name = basetype(FSA)
-    :($name{$(FSA.parameters[1]), T, $(FSA.parameters[3:end]...)})
-end
-@generated function similar{FSA <: FixedVectorNoTuple, T}(::Type{FSA}, ::Type{T})
-    name = basetype(FSA)
-    :($name{T, $(FSA.parameters[3:end]...)})
+if VERSION < v"0.5.0-dev+698"
+    macro pure(ex)
+        esc(ex)
+    end
+else
+    import Base: @pure
```

Comment:
Do anyone have any comments on this `@pure` stuff BTW?  It's a little disturbing that `@pure` is not even exported from Base, so I'm still a bit iffy about it.